pkg/host/internal/sriov/sriov.go

```diff
@@ -467,76 +502,184 @@ func (s *sriov) ConfigSriovInterfaces(storeManager store.ManagerInterface,
 				}
-				break
+				if skip {
+					break
+				}
+				toBeConfigured = append(toBeConfigured, interfaceToConfigure{iface: iface, ifaceStatus: ifaceStatus})
 			}
 		}
+
 		if !configured && ifaceStatus.NumVfs > 0 {
-			if skip := pfsToConfig[ifaceStatus.PciAddress]; skip {
-				continue
+			if skip := pfsToConfig[ifaceStatus.PciAddress]; !skip {
+				toBeResetted = append(toBeResetted, ifaceStatus)
 			}
+		}
+	}
+	return toBeConfigured, toBeResetted, nil
+}
 
-			// load the PF info
-			pfStatus, exist, err := storeManager.LoadPfsStatus(ifaceStatus.PciAddress)
-			if err != nil {
-				log.Log.Error(err, "SyncNodeState(): failed to load info about PF status for device",
-					"address", ifaceStatus.PciAddress)
-				return err
+func (s *sriov) configSriovInterfacesInParallel(storeManager store.ManagerInterface, interfaces []interfaceToConfigure) error {
+	log.Log.V(2).Info("ConfigSriovInterfacesInParallel(): start sriov configuration")
+
+	// TODO(e0ne): store all errors in SriovNetworkNodeState
+	var result error
+	errChannel := make(chan error)
+	interfacesToConfigure := 0
+	for ifaceIndex, iface := range interfaces {
+		interfacesToConfigure += 1
+		go func(iface *interfaceToConfigure) {
+			if err := s.ConfigSriovDevice(&iface.iface, &iface.ifaceStatus); err != nil {
+				log.Log.Error(err, "ConfigSriovInterfacesInParallel(): fail to configure sriov interface. resetting interface.", "address", iface.iface.PciAddress)
+				if iface.iface.ExternallyManaged {
+					log.Log.Info("ConfigSriovInterfacesInParallel(): skipping device reset as the nic is marked as externally created")
+				} else {
+					if resetErr := s.ResetSriovDevice(iface.ifaceStatus); resetErr != nil {
+						log.Log.Error(resetErr, "ConfigSriovInterfacesInParallel(): failed to reset on error SR-IOV interface")
+					}
+				}
+				errChannel <- err
+			} else {
+				errChannel <- nil
 			}
+		}(&interfaces[ifaceIndex])
+		// Save the PF status to the host
+		err := storeManager.SaveLastPfAppliedStatus(&iface.iface)
+		if err != nil {
+			log.Log.Error(err, "ConfigSriovInterfacesInParallel(): failed to save PF applied config to host")
+			return err
+		}
+		break
+	}
 
-			if !exist {
-				log.Log.Info("SyncNodeState(): PF name with pci address has VFs configured but they weren't created by the sriov operator. Skipping the device reset",
-					"pf-name", ifaceStatus.Name,
-					"address", ifaceStatus.PciAddress)
-				continue
+	for i := 0; i < interfacesToConfigure; i++ {
+		errMsg := <-errChannel
+		result = errors.Join(result, errMsg)
+	}
+	if result != nil {
+		log.Log.Error(result, "ConfigSriovInterfacesInParallel(): fail to configure sriov interface")
+		return result
+	}
+	log.Log.V(2).Info("ConfigSriovInterfacesInParallel(): sriov configuration finished")
+	return nil
+}
+
+func (s *sriov) resetSriovInterfacesInParallel(storeManager store.ManagerInterface, interfaces []sriovnetworkv1.InterfaceExt) error {
+	var result error
+	errChannel := make(chan error)
```

Comment:
you could initialize this with capacity of `len(interfaces)` so that go routines don block on errChannel.
not a must, up to u.